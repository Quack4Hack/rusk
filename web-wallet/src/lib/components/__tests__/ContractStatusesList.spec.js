import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { ContractStatusesList } from "..";

describe("ContractStatusesList", () => {
  const baseProps = {
    statuses: [
      {
        label: "Spendable",
        value: "99,899.999724165",
      },
      {
        label: "Total Locked",
        value: "1,000.000000000",
      },
      {
        label: "Rewards",
        value: "99,288.000000000",
      },
    ],
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("should render the `ContractStatusesList` component", () => {
    const { container } = render(ContractStatusesList, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should be able to render the component without items", () => {
    const props = {
      ...baseProps,
      statuses: [],
    };
    const { container } = render(ContractStatusesList, {
      ...baseOptions,
      props,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
