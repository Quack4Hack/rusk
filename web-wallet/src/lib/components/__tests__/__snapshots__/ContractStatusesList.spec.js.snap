// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContractStatusesList > should be able to render the component without items 1`] = `
<div
  class="contract-statuses svelte-1up3579"
>
  <dl
    class="contract-statuses__list svelte-1up3579"
  />
   
</div>
`;

exports[`ContractStatusesList > should render the \`ContractStatusesList\` component 1`] = `
<div
  class="contract-statuses svelte-1up3579"
>
  <dl
    class="contract-statuses__list svelte-1up3579"
  >
    <dt
      class="contract-statuses__label svelte-1up3579"
    >
      Spendable
       
    </dt>
    <dd
      class="contract-statuses__value svelte-1up3579"
    >
      <span>
        99,899.999724165
      </span>
       
      <svg
        class="dusk-icon dusk-icon--size--default contract-statuses__icon"
        data-tooltip-id="main-tooltip"
        data-tooltip-text="DUSK"
        role="graphics-symbol"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
        />
      </svg>
      
       
    </dd>
    <dt
      class="contract-statuses__label svelte-1up3579"
    >
      Total Locked
       
    </dt>
    <dd
      class="contract-statuses__value svelte-1up3579"
    >
      <span>
        1,000.000000000
      </span>
       
      <svg
        class="dusk-icon dusk-icon--size--default contract-statuses__icon"
        data-tooltip-id="main-tooltip"
        data-tooltip-text="DUSK"
        role="graphics-symbol"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
        />
      </svg>
      
       
    </dd>
    <dt
      class="contract-statuses__label svelte-1up3579"
    >
      Rewards
       
    </dt>
    <dd
      class="contract-statuses__value svelte-1up3579"
    >
      <span>
        99,288.000000000
      </span>
       
      <svg
        class="dusk-icon dusk-icon--size--default contract-statuses__icon"
        data-tooltip-id="main-tooltip"
        data-tooltip-text="DUSK"
        role="graphics-symbol"
        viewBox="0 0 24 24"
      >
        <path
          d="M12.335 0.005a12.186 12.186 0 0 0 -2.58 0.204C4.2 1.259 0 6.14 0 11.999 0 17.859 4.203 22.74 9.759 23.79c0.726 0.138 1.476 0.21 2.241 0.21 6.789 0 12.261 -5.64 11.991 -12.49C23.741 5.249 18.599 0.173 12.335 0.005Zm0.207 21.58c-0.204 0.012 -0.336 -0.222 -0.21 -0.384 1.974 -2.541 3.153 -5.733 3.153 -9.202 0 -3.468 -1.176 -6.663 -3.153 -9.202 -0.126 -0.162 0.003 -0.396 0.207 -0.384 5.052 0.282 9.06 4.465 9.06 9.586 0 5.121 -4.005 9.304 -9.057 9.586Z"
        />
      </svg>
      
       
    </dd>
  </dl>
   
</div>
`;
